### DISCUSSION TITLE
How to manage manual migrations and prisma migrations in production environment?

### DISCUSSION THREAD

**Discussion by florianmartens** [prisma/prisma#2770] [Q&A] [answered]
Hi,

so, I'm at an impasse and I'm not sure if it's my lack of skill or a question lots of others have too.

I generally enjoy Prisma and would like to use it in my current project. Unfortunately, there are still some gaps in the Prisma schema (e. g. supporting cascading deletes) and the migration system. In order to get things to work, we have to write some manual SQL migrations. So we end up with Prisma migrations + manual migrations that have to be applied to a production environment.

I'm currently not sure how to orchestrate between the two. Let's say my first Prisma migration creates the table Users. Afterward, I need to apply a manual migration adding some behavior and then continue with some of the auto-generated Prisma migrations.

Are there any resources on how to do this in a production environment?

Thanks!

**Comment by Jolg42** [comment, marked as answer]
Hi!

@florianmartens I think Migrations Hooks would help here, what do you think?

It's not implemented yet, the tracking issue is here https://github.com/prisma/migrate/issues/195 (it has a link the detailed specification too)

Is that exactly what you need? We're open to hear your feedback 😃

**Reply by florianmartens** [reply]
I'll have a look at it. It would certainly be helpful to just be able to create migration files that execute plain SQL - at least until the migration system is a bit more mature.

**Comment by ludralph** [comment]
Hi @florianmartens,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
